lib/cuckoo/core/guest.py

```diff
@@ -343,21 +343,23 @@ def upload_analyzer(self, monitor):
         self.post("/extract", files={"zipfile": zip_data}, data=data)
 
     def add_config(self, options):
-        """Upload the analysis.conf for this task to the Virtual Machine."""
-        config = [
-            "[analysis]",
-        ]
+        """Upload the analysis.json for this task to the Virtual Machine."""
+        config = {"analysis" : {}}
         for key, value in options.items():
-            # Encode datetime objects the way xmlrpc encodes them.
-            if isinstance(value, datetime.datetime):
-                config.append("%s = %s" % (key, value.strftime("%Y%m%dT%H:%M:%S")))
+            # Json can only serialize these types
+            if type(value) in [dict, list, str, int, float, bool]:
+                config["analysis"][key] = value
             else:
-                config.append("%s = %s" % (key, value))
+                try:
+                    value = value.__str__()
+                except AttributeError:
+                    value = ""
+                config["analysis"][key] = value
 
         data = {
-            "filepath": os.path.join(self.analyzer_path, "analysis.conf"),
+            "filepath": os.path.join(self.analyzer_path, "analysis.json"),
         }
-        self.post("/store", files={"file": "\n".join(config)}, data=data)
+        self.post("/store", files={"file": json.dumps(config)}, data=data)
```

Comment:
You're communicating with the new Cuckoo Agent here, but made changes for the Old Agent above? Has this functionality been tested?